Modules/readline.c

```diff
@@ -1026,7 +1026,19 @@ on_startup_hook(void)
 {
     int r;
     PyGILState_STATE gilstate = PyGILState_Ensure();
-    r = on_hook(readlinestate_global->startup_hook);
+    PyObject* mod = PyState_FindModule(&readlinemodule);
+    if (mod == NULL) {
+        PyGILState_Release(gilstate);
+        if (!PyErr_Occurred()) {
+            PyErr_SetString(PyExc_ImportError, "readline import initialization failure during startup hook");
+        }
```

Comment:
```suggestion
        if (!PyErr_Occurred()) {
            PyErr_SetString(PyExc_ImportError, "readline import initialization failure during startup hook");
        }
        PyGILState_Release(gilstate);
```

Reply:
Alternately, change line 1027 to `int r = -1;`, add an `error:` label before the last two lines (i.e. before line 1042), and then:

```suggestion
        if (!PyErr_Occurred()) {
            PyErr_SetString(PyExc_ImportError, "readline import initialization failure during startup hook");
        }
        goto error;
```

I probably wouldn't bother with that, though, since there's currently only the one branch that would use the label.